core/runtime/src/main/java/io/quarkus/runtime/configuration/ConfigUtils.java

```diff
@@ -252,18 +250,25 @@ public String getName() {
      */
     static class BuildTimeSysPropConfigSource extends SysPropConfigSource {
         public Map<String, String> getProperties() {
-            Map<String, String> output = new TreeMap<>();
-            for (Map.Entry<Object, Object> entry : System.getProperties().entrySet()) {
-                String key = (String) entry.getKey();
-                if (key.startsWith("quarkus.")) {
-                    output.put(key, entry.getValue().toString());
-                }
-            }
-            return output;
+            BuildTimeSysPropMapProducer buildTimeSysPropMapProducer = new BuildTimeSysPropMapProducer();
```

Comment:
With this change we fewer allocations as is evident by looking up how `System.getProperties().entrySet()` is implemented